### DISCUSSION TITLE
OpenSSL 3.4.0 Alpha1 was released

### DISCUSSION THREAD

**Discussion by t8m** [openssl/openssl#25391] [Announcements]
We are pleased to announce the immediate availability of OpenSSL 3.4.0 Alpha 1. This release incorporates a number of new features, most notably:

  * Support for directly fetched composite signature algorithms such as
    RSA-SHA2-256 including new API functions

  * FIPS indicators support in the FIPS provider and various updates of the FIPS
    provider required for future FIPS 140-3 validations

  * Implementation of RFC 9579 (PBMAC1) in PKCS#12

  * An optional additional random seed source RNG `JITTER` using a statically
    linked jitterentropy library

  * New options `-not_before` and `-not_after` for explicit setting start and
    end dates of certificates created with the `req` and `x509` apps

  * Support for integrity-only cipher suites TLS_SHA256_SHA256 and
    TLS_SHA384_SHA384 in TLS 1.3, as defined in RFC 9150

  * Support for requesting CRL in CMP

  * Support for additional X.509v3 extensions related to Attribute Certificates

  * Initial Attribute Certificate (RFC 5755) support

  * Possibility to customize ECC groups initialization to use precomputed values
    to save CPU time and use of this feature by the P-256 implementation

We would like to thank all of our users and communities for their continued use and support of OpenSSL. This release would not be possible without the innumerate bug reports, pull requests, code reviews and feedback we continue to receive from our community.

OpenSSL 3.4.0 Alpha 1 can be downloaded here: https://github.com/openssl/openssl/releases/tag/openssl-3.4.0-alpha1

Please download and test it.

**Comment by makr** [comment]
Will 3.4 become the new LTS version?

**Reply by t8m** [reply]
No, not likely.

**Comment by darkage2** [comment]
What is the planned release date for this version?

The previous ones proved to be unusable for us, due to the performance loss. The improvements are pretty huge in this.
In our testing, DTLS hasndshakes/s using AES128-SHA256 had 20%+ loss with 3.0, 3.1, 3.2 and 3.3 compared to 1.1.1w, while 3.4-dev's largest was 10%.
With DHE-RSA-AES256-SHA256 and 2048bit DH params, the previous 95% loss became an improvement of almost 50%. It's basically on par with our old measurements for 1024bit DH.

**Reply by t8m** [reply]
The planned release date for 3.4.0 is in the middle of October.

**Comment by pgarba** [comment]
Are you planning to back port the FIPS indicators and updates to the 3.1 branch ?
We started the certification based on 3.1.x and I see this is also a requirement in the FTP

**Reply by paulidale** [reply]
No.

Intertek Acumen Security, who are doing the OpenSSL 3.1 validation, believe they can justify implicit indicators for 3.1.
I'm not entirely convinced but I'm not an expert.

Going forward explicit indicators seem like they will be mandatory, so we added them in 3.4.

**Reply by pgarba** [reply]
So what is your plan for OpenSSL FIPS 140-3 certification ? You restart the certification on the 3.4 branch ?

**Reply by paulidale** [reply]
Ask the OpenSSL project leaders.  I'm no longer one of them.

**Reply by pgarba** [reply]
Do you have a contact person for me ?

**Reply by paulidale** [reply]
https://openssl-library.org/about/management/index.html

**Comment by nhorman** [comment]
we are working with our lab to look at options for our 3.1.2 provider to provide compliant indicator methods there, and will decide if a backport is needed

**Reply by pgarba** [reply]
do you have a timeline when the decision will be done ? 
Just thinking how to continue on our side.

**Reply by nhorman** [reply]
That really depends on the response from CMVP when they review our validation submission.  If they indicate that our implicit indicator assertions aren't acceptable, we need to do the backport

**Reply by pgarba** [reply]
Thanks for your answer.
 I will check with our lab (Lightship) how to continue on our side then.
